### PR TITLE
[ci-visibility] Fix CI Visibility Spec Tests

### PR DESCRIPTION
**Pull request by juan-fernandez**
## Description
* Fix branch extraction in gitlab.
* Add stage and job name in azure pipelines.
* Add `usersupplied.json` as fixture (for user defined git metadata).
* Fix spec tests.

Context: https://github.com/DataDog/dd-trace-rb/pull/1687

## Checklist
- [ ] Added to the correct milestone.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Comment by brettlangdon** [comment]
@juan-fernandez can you add a changelog entry(ies) for this PR?

File: tests/tracer/fixtures/ci/appveyor.json [hunk 1]
```diff
@@ -7,9 +7,9 @@
       "APPVEYOR_BUILD_NUMBER": "appveyor-pipeline-number",
       "APPVEYOR_REPO_BRANCH": "master",
       "APPVEYOR_REPO_COMMIT": "appveyor-repo-commit",
-      "APPVEYOR_REPO_COMMIT_MESSAGE_EXTENDED": "appveyor-commit-message",
       "APPVEYOR_REPO_COMMIT_AUTHOR": "appveyor-commit-author-name",
       "APPVEYOR_REPO_COMMIT_AUTHOR_EMAIL": "appveyor-commit-author-email@datadoghq.com",
+      "APPVEYOR_REPO_COMMIT_MESSAGE_EXTENDED": "appveyor-commit-message",
```
Comment (brettlangdon) [inline tests/tracer/fixtures/ci/appveyor.json:12]: it doesn't look like anything is changing in this file? seems like moving these keys to all be alphabetical order?

Comment (brettlangdon) [reply tests/tracer/fixtures/ci/appveyor.json:12]: Same with the other files, looks like some new test cases, alphabetical sorting of the keys.

The only ones that are really "different" are Gitlab CI_COMMIT_BRANCH -> CI_COMMIT REF_NAME, and Azure adding stage and job name ?

Comment (juan-fernandez) [reply tests/tracer/fixtures/ci/appveyor.json:12]: > The only ones that are really "different" are Gitlab CI_COMMIT_BRANCH -> CI_COMMIT REF_NAME, and Azure adding stage and job name ?
that's correct. Sorry about that, we have unstable sorting in the spec repo

**Comment by juan-fernandez** [comment]
> @juan-fernandez can you add a changelog entry(ies) for this PR?

sure, will do 👍

File: tests/tracer/fixtures/ci/appveyor.json (hunk 1 above)
Comment (brettlangdon) [reply tests/tracer/fixtures/ci/appveyor.json:12]: 👍🏻 no worries, I assume these are copy/past from a source, so as long as we know what needs review then all good

**Comment by juan-fernandez** [comment]
> > @juan-fernandez can you add a changelog entry(ies) for this PR?
> 
> sure, will do 👍

added it!

**Comment by juan-fernandez** [comment]
sanic tests are failing, but they seem to be flakes

**Comment by Kyle-Verhoog** [comment]
@juan-fernandez yeah sanic is failing unrelated, due to a new release. Working on a fix for it!

**Comment by juan-fernandez** [comment]
does it look good otherwise @brettlangdon ?

**Comment by juan-fernandez** [comment]
sanic tests are passing now 👍

File: releasenotes/notes/update-ci-metadata-extraction-1f5c1b2b0da14287.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+upgrade:
+  - |
+    Only for CI Visibility (`pytest` integration): Extract stage and job name from environment data in Azure Pipelines.
+fixes:
+  - |
+    Only for CI Visibility (`pytest` integration): Fix extraction of branch in GitLab CI.
```
Comment (brettlangdon) [inline releasenotes/notes/update-ci-metadata-extraction-1f5c1b2b0da14287.yaml:7]:
```suggestion
features:
  - |
    Only for CI Visibility (``pytest`` integration): Extract stage and job name from environment data in Azure Pipelines.
fixes:
  - |
    Only for CI Visibility (``pytest`` integration): Fix extraction of branch in GitLab CI.
```

**Review by brettlangdon — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

**Pull request merged**